client/src/components/Charts/datepicker.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import DatePicker from 'react-datepicker';
+import PropTypes from 'prop-types';
+
+import 'react-datepicker/dist/react-datepicker.css';
+
+const CustomDatePicker = (props) => {
+  const { placeholder, date, handleDateChange } = props;
+  return (
+    <div>
+      <div>{placeholder}</div>
+      <DatePicker
+        selected={date}
+        onChange={handleDateChange}
+        maxDate={new Date()}
```

Comment:
You could move the `new Date()` to the top of the file, in a const, instead of putting it here inside the component render.  Then, it won't create a new date instance every time the component re-renders (which would happen every time the user changes the date).